nominal/core.py

```diff
@@ -879,6 +900,42 @@ def _search_datasets(self) -> Iterable[Dataset]:
         for ds in response.results:
             yield Dataset._from_conjure(self, ds)
 
+    def _get_user_rid_from_email(self, user_email: str) -> str:
+        request = authentication_api.SearchUsersRequest(
+            query=authentication_api.SearchUsersQuery(
+                exact_match=user_email,
+            )
+        )
+        response = self._authentication_client.search_users_v2(self._auth_header, request)
+        if len(response.results) == 0:
+            raise ValueError(f"user {user_email!r} not found")
+        if len(response.results) > 1:
+            raise ValueError(f"found multiple users with email {user_email!r}")
+        return response.results[0].rid
+
+    def get_checklist(self, rid: str) -> Checklist:
+        response = self._checklist_api_client.get(self._auth_header, rid)
+        return Checklist._from_conjure(self, response)
+
+    def checklist_builder(
+        self,
+        name: str,
+        assignee_email: str,
```

Comment:
Let's make this optional and use the whoami endpoint in the event that an email is not provided.